library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -209,6 +226,59 @@ class Client() {
         publishUserContact(legacy = true)
     }
 
+    fun importConversation(conversationData: ByteArray): Conversation {
+        val gson = GsonBuilder().create()
+        val v2Export = gson.fromJson(
+            conversationData.toString(StandardCharsets.UTF_8),
+            ConversationV2Export::class.java
+        )
+        try {
+            return importV2Conversation(export = v2Export)
+        } catch (e: java.lang.Exception) {
+            val v1Export = gson.fromJson(
+                conversationData.toString(StandardCharsets.UTF_8),
+                ConversationV1Export::class.java
+            )
+            try {
+                return importV1Conversation(export = v1Export)
+            } catch (e: java.lang.Exception) {
+                throw XMTPException("Invalid input data", e)
+            }
+        }
+    }
+
+    fun importV2Conversation(export: ConversationV2Export): Conversation {
+        val keyMaterial = Base64.decode(export.keyMaterial)
+        return Conversation.V2(
+            ConversationV2(
+                topic = export.topic,
+                keyMaterial = keyMaterial,
+                context = InvitationV1ContextBuilder.buildFromConversation(
+                    conversationId = export.context?.conversationId ?: "",
+                    metadata = export.context?.metadata ?: mapOf()
+                ),
+                peerAddress = export.peerAddress,
+                client = this,
+                header = SealedInvitationHeaderV1.newBuilder().build()
+            )
+        )
+    }
+
+    fun importV1Conversation(export: ConversationV1Export): Conversation {
+        val sentAt = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            Date.from(Instant.parse(export.createdAt))
+        } else {
+            Date()
+        }
```

Comment:
All of these APIs that allow parsing a UTC date string into a date require API level 26 we currently are targeting 22. This isn't the first time I've run into issues and had to do work around to be able to support 22.
In this case if someone is using an API below 26 and import a V1 conversation (rare) all the conversations will be incorrectly dated to the date that they imported and not the original date. I tried a bunch of different things but wasn't able to make anything work for below 26. Open to suggestions.